generic/Storage.c

```diff
@@ -39,6 +39,27 @@ static int torch_Storage_(new)(lua_State *L)
       lua_pop(L, 1);
     }
   }
+  else if(lua_type(L, index) == LUA_TUSERDATA)
+  {
+    THStorage *src = luaT_checkudata(L, index, torch_Storage);
+    if (src->flag & TH_STORAGE_VIEW) {
+      // don't allow chained views
+      src = src->view;
```

Comment:
This doesn't seem very intuitive. If you don't want to support chained views, why no throw an error? What is the problem with chained views in the first place?